plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakePartitionsTable.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.trino.plugin.deltalake;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.plugin.deltalake.transactionlog.AddFileEntry;
+import io.trino.plugin.deltalake.transactionlog.MetadataEntry;
+import io.trino.plugin.deltalake.transactionlog.TableSnapshot;
+import io.trino.plugin.deltalake.transactionlog.TransactionLogAccess;
+import io.trino.plugin.deltalake.util.PageListBuilder;
+import io.trino.spi.Page;
+import io.trino.spi.TrinoException;
+import io.trino.spi.connector.ColumnMetadata;
+import io.trino.spi.connector.ConnectorPageSource;
+import io.trino.spi.connector.ConnectorSession;
+import io.trino.spi.connector.ConnectorTableMetadata;
+import io.trino.spi.connector.ConnectorTransactionHandle;
+import io.trino.spi.connector.EmptyPageSource;
+import io.trino.spi.connector.FixedPageSource;
+import io.trino.spi.connector.SchemaTableName;
+import io.trino.spi.connector.SystemTable;
+import io.trino.spi.predicate.TupleDomain;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.trino.spi.type.VarcharType.VARCHAR;
+import static java.util.Objects.requireNonNull;
+
+public class DeltaLakePartitionsTable
+        implements SystemTable
+{
+    private final SchemaTableName tableName;
+    private final String tableLocation;
+    private final TransactionLogAccess transactionLogAccess;
+    private final ConnectorTableMetadata tableMetadata;
+    private final List<DeltaLakeColumnHandle> partitionColumns;
+
+    public DeltaLakePartitionsTable(
+            SchemaTableName tableName,
+            String tableLocation,
+            TransactionLogAccess transactionLogAccess,
+            List<DeltaLakeColumnHandle> partitionColumns)
+    {
+        this.tableName = requireNonNull(tableName, "tableName is null");
+        this.tableLocation = requireNonNull(tableLocation, "tableLocation is null");
+        this.transactionLogAccess = requireNonNull(transactionLogAccess, "transactionLogAccess is null");
+        this.partitionColumns = requireNonNull(partitionColumns, "partitionColumns is null");
+
+        ImmutableList.Builder<ColumnMetadata> columns = ImmutableList.builder();
+        partitionColumns.forEach(column -> columns.add(new ColumnMetadata(column.getBaseColumnName(), VARCHAR)));
```

Comment:
i don't follow the reasoning behind this constraint - why not using the actual type of the partition column?

Reply:
I initially tried to do that, but there were issues with casting back to the original type as `AddFileEntry.getCanonicalPartitionValues()` returns a string represenation of the value

Reply:
It should be the original type. Also, we need to discuss the result columns (Hive or Iceberg)
```sql
-- Hive
select * from "test$partitions";
 b
---
 1

-- Iceberg
select * from "test$partitions";
 partition | record_count | file_count | total_size |                       data
-----------+--------------+------------+------------+--------------------------------------------------
 {b=1}     |            1 |          1 |        317 | {a={min=1, max=1, null_count=0, nan_count=NULL}}
```

Reply:
I'll try to see if I can get the types correct 👍 

Good point on format, I've only used the `hive` and `delta` connectors in practice so went with the original Hive format. The Iceberg format has a lot more info, but I'm not sure how best to materialize that from the transaction log.